src/server/game/Battlegrounds/Zones/BattlegroundWS.cpp

```diff
@@ -518,6 +518,14 @@ void BattlegroundWS::EventPlayerClickedOnFlag(Player* player, GameObject* target
             PlaySoundToAll(BG_WS_SOUND_FLAG_RETURNED);
             UpdatePlayerScore(player, SCORE_FLAG_RETURNS, 1);
             _bothFlagsKept = false;
+
+            //check if HORDE flag is in capture zone, if so, capture it
+            if (Player* player = ObjectAccessor::FindPlayer(GetFlagPickerGUID(TEAM_HORDE)))
```

Comment:
Because this function runs in map update context, using `ObjectAccessor::FindPlayer` is unsafe, replace it with `ObjectAccessor::GetPlayer(GetBgMap(), GetFlagPickerGUID(TEAM_HORDE))` (and do the same for alliance code)

Reply:
Also, correct spelling: // Check Horde flag if it is in capture zone; if so, capture it

Reply:
> Also, correct spelling: // Check Horde flag if it is in capture zone; if so, capture it

Fixed :)